File: packages/js-sdk/src/regex/detectors/ee/personal-code.ts
// Estonian Personal Code (Isikukood) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { eePersonalCodeChecksum } from '../../validators'

class EePersonalCodeDetector extends RegexDetector {
  entityType = EntityType.EE_PERSONAL_CODE
  score = 0.85
  contextKeywords = [
    'isikukood',
    'personal code',
    'identity code',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b[1-6]\d{10}\b/g
  validator = eePersonalCodeChecksum
}

registerRegion('ee', EePersonalCodeDetector)

export { EePersonalCodeDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/ee/personal_code.py
"""Estonian Personal Code (Isikukood) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import ee_personal_code_checksum


@register_region("ee")
class PersonalCodeDetector(RegexDetector):
    entity_type = EntityType.EE_PERSONAL_CODE
    score = 0.85
    context_keywords = [
        "isikukood", "personal code", "identity code",
    ]
    context_required = True
    context_window = 50

    # Gender digit (1-6) followed by 10 digits
    pattern = re.compile(
        r"\b[1-6]\d{10}\b"
    )
    validator = staticmethod(ee_personal_code_checksum)
